judge/admin/problem.py

```diff
@@ -248,3 +249,23 @@ def construct_change_message(self, request, form, *args, **kwargs):
         if form.cleaned_data.get('change_message'):
             return form.cleaned_data['change_message']
         return super(ProblemAdmin, self).construct_change_message(request, form, *args, **kwargs)
+
+
+class ProblemPointsVoteAdmin(admin.ModelAdmin):
+    list_display = ('points', 'voter', 'problem', 'vote_time')
+    search_fields = ('voter__user__username', 'problem__code', 'problem__name')
+    readonly_fields = ('voter', 'problem', 'vote_time')
+
+    def get_queryset(self, request):
+        return ProblemPointsVote.objects.filter(problem__in=Problem.get_editable_problems(request.user))
+
+    def has_add_permission(self, request):
+        return False
+
+    def has_change_permission(self, request, obj=None):
+        if obj is None:
+            return request.user.has_perm('judge.edit_own_problem')
```

Comment:
Remind me why this is `judge.edit_own_problem`?

Reply:
https://docs.djangoproject.com/en/2.2/ref/contrib/admin/#django.contrib.admin.ModelAdmin.has_change_permission

`True` means that the current user can edit at least 1 vote.